bug: The android app crashes with version 5.1.0

## Bug Report

### Capacitor Version

```
tried 5.4.1. and 5.3.0
```

Tested using Nexus 10 API 33 emulator.

This is from Android studio Logcat 
```
Caused by: java.security.ProviderException: Keystore key generation failed
    at android.security.keystore2.AndroidKeyStoreKeyGeneratorSpi.engineGenerateKey(AndroidKeyStoreKeyGeneratorSpi.java:413)
    at javax.crypto.KeyGenerator.generateKey(KeyGenerator.java:612)
    at ee.forgr.biometric.NativeBiometric.generateKey(NativeBiometric.java:383)
    at ee.forgr.biometric.NativeBiometric.generateKey(NativeBiometric.java:357)
    at ee.forgr.biometric.NativeBiometric.getKey(NativeBiometric.java:396)
    at ee.forgr.biometric.NativeBiometric.encryptString(NativeBiometric.java:319)
    at ee.forgr.biometric.NativeBiometric.setCredentials(NativeBiometric.java:202)
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.getcapacitor.PluginHandle.invoke(PluginHandle.java:138) 
    at com.getcapacitor.Bridge.lambda$callPluginMethod$0(Bridge.java:774) 
    at com.getcapacitor.Bridge.$r8$lambda$ehFTi5f4HhVNFKTbCKAYDkpQYRA(Unknown Source:0) 
    at com.getcapacitor.Bridge$$ExternalSyntheticLambda3.run(Unknown Source:8) 
    at android.os.Handler.handleCallback(Handler.java:942) 
    at android.os.Handler.dispatchMessage(Handler.java:99) 
    at android.os.Looper.loopOnce(Looper.java:201) 
    at android.os.Looper.loop(Looper.java:288) 
    at android.os.HandlerThread.run(HandlerThread.java:67) 
Caused by: android.security.KeyStoreException: System error (internal Keystore code: 4 message: In generate_key.
```

It works when downgraded to 5.0.1.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,15 +2,16 @@ ext {
     junitVersion = project.hasProperty('junitVersion') ? rootProject.ext.junitVersion : '4.13.2'
     androidxJunitVersion = project.hasProperty('androidxJunitVersion') ? rootProject.ext.androidxJunitVersion : '1.1.5'
     androidxEspressoCoreVersion = project.hasProperty('androidxEspressoCoreVersion') ? rootProject.ext.androidxEspressoCoreVersion : '3.5.1'
+    androidxAppCompatVersion = project.hasProperty('androidxAppCompatVersion') ? rootProject.ext.androidxAppCompatVersion : '1.6.1'
 }
 
 buildscript {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:8.0.0'
+        classpath 'com.android.tools.build:gradle:8.0.2'
     }
 }
 
@@ -39,20 +40,19 @@ android {
 
 repositories {
     google()
-    jcenter()
     mavenCentral()
 }
 
 
 dependencies {
-    implementation 'androidx.biometric:biometric:1.0.1'
+    implementation 'androidx.biometric:biometric:1.1.0'
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation project(':capacitor-android')
-    implementation 'androidx.appcompat:appcompat:1.1.0'
-    implementation 'com.google.android.material:material:1.1.0'
-    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation 'androidx.navigation:navigation-fragment:2.0.0'
-    implementation 'androidx.navigation:navigation-ui:2.0.0'
+    implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
+    implementation 'com.google.android.material:material:1.9.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'androidx.navigation:navigation-fragment:2.5.3'
+    implementation 'androidx.navigation:navigation-ui:2.5.3'
     testImplementation "junit:junit:$junitVersion"
     androidTestImplementation "androidx.test.ext:junit:$androidxJunitVersion"
     androidTestImplementation "androidx.test.espresso:espresso-core:$androidxEspressoCoreVersion"

--- a/android/src/main/java/ee/forgr/biometric/AuthActivity.java
+++ b/android/src/main/java/ee/forgr/biometric/AuthActivity.java
@@ -4,15 +4,9 @@ import android.content.Intent;
 import android.os.Build;
 import android.os.Bundle;
 import android.os.Handler;
-import android.util.Log;
-import android.view.View;
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
-import androidx.appcompat.widget.Toolbar;
-import androidx.biometric.BiometricConstants;
 import androidx.biometric.BiometricPrompt;
-import com.google.android.material.floatingactionbutton.FloatingActionButton;
-import com.google.android.material.snackbar.Snackbar;
 import ee.forgr.biometric.capacitornativebiometric.R;
 import java.util.concurrent.Executor;
 
@@ -136,29 +130,29 @@ public class AuthActivity extends AppCompatActivity {
    * Convert Auth Error Codes to plugin expected Biometric Auth Errors (in README.md)
    * This way both iOS and Android return the same error codes for the same authentication failure reasons.
    * !!IMPORTANT!!: Whenever this is modified, check if similar function in iOS Plugin.swift needs to be modified as well
-   * @see https://developer.android.com/reference/androidx/biometric/BiometricPrompt#constants
+   * @see <a href="https://developer.android.com/reference/androidx/biometric/BiometricPrompt#constants">...</a>
    * @return BiometricAuthError
    */
   public static int convertToPluginErrorCode(int errorCode) {
     switch (errorCode) {
-      case BiometricConstants.ERROR_HW_UNAVAILABLE:
-      case BiometricConstants.ERROR_HW_NOT_PRESENT:
+      case BiometricPrompt.ERROR_HW_UNAVAILABLE:
+      case BiometricPrompt.ERROR_HW_NOT_PRESENT:
         return 1;
-      case BiometricConstants.ERROR_LOCKOUT_PERMANENT:
+      case BiometricPrompt.ERROR_LOCKOUT_PERMANENT:
         return 2;
-      case BiometricConstants.ERROR_NO_BIOMETRICS:
+      case BiometricPrompt.ERROR_NO_BIOMETRICS:
         return 3;
-      case BiometricConstants.ERROR_LOCKOUT:
+      case BiometricPrompt.ERROR_LOCKOUT:
         return 4;
       // Authentication Failure (10) Handled by `onAuthenticationFailed`.
       // App Cancel (11), Invalid Context (12), and Not Interactive (13) are not valid error codes for Android.
-      case BiometricConstants.ERROR_NO_DEVICE_CREDENTIAL:
+      case BiometricPrompt.ERROR_NO_DEVICE_CREDENTIAL:
         return 14;
-      case BiometricConstants.ERROR_TIMEOUT:
-      case BiometricConstants.ERROR_CANCELED:
+      case BiometricPrompt.ERROR_TIMEOUT:
+      case BiometricPrompt.ERROR_CANCELED:
         return 15;
-      case BiometricConstants.ERROR_USER_CANCELED:
-      case BiometricConstants.ERROR_NEGATIVE_BUTTON:
+      case BiometricPrompt.ERROR_USER_CANCELED:
+      case BiometricPrompt.ERROR_NEGATIVE_BUTTON:
         return 16;
       default:
         return 0;

--- a/android/src/main/java/ee/forgr/biometric/NativeBiometric.java
+++ b/android/src/main/java/ee/forgr/biometric/NativeBiometric.java
@@ -14,7 +14,6 @@ import android.security.keystore.KeyProperties;
 import android.security.keystore.StrongBoxUnavailableException;
 import android.util.Base64;
 import androidx.activity.result.ActivityResult;
-import androidx.biometric.BiometricConstants;
 import androidx.biometric.BiometricManager;
 import com.getcapacitor.JSObject;
 import com.getcapacitor.Plugin;
@@ -124,7 +123,7 @@ public class NativeBiometric extends Plugin {
     // @see https://developer.android.com/reference/androidx/biometric/BiometricManager#canAuthenticate(int)
     boolean fallbackAvailable = useFallback && this.deviceHasCredentials();
     if (useFallback && !fallbackAvailable) {
-      canAuthenticateResult = BiometricConstants.ERROR_NO_DEVICE_CREDENTIAL;
+      canAuthenticateResult = BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE;
     }
 
     boolean isAvailable =
@@ -496,6 +495,10 @@ public class NativeBiometric extends Plugin {
     KeyguardManager keyguardManager = (KeyguardManager) getActivity()
       .getSystemService(Context.KEYGUARD_SERVICE);
     // Can only use fallback if the device has a pin/pattern/password lockscreen.
-    return keyguardManager.isDeviceSecure();
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      return keyguardManager.isDeviceSecure();
+    } else {
+      return keyguardManager.isKeyguardSecure();
+    }
   }
 }

